numpy/core/_internal.py

```diff
@@ -843,11 +843,13 @@ def _ufunc_doc_signature_formatter(ufunc):
         ", order='K'"
         ", dtype=None"
         ", subok=True"
-        "[, signature"
-        ", extobj]"
     )
+
+    # NOTE: gufuncs may or may not support the `axis` parameter
     if ufunc.signature is None:
-        kwargs = ", where=True" + kwargs
+        kwargs = f", where=True{kwargs}[, signature, extobj]"
+    else:
+        kwargs += "[, signature, extobj, axes, axis]"
```

Comment:
Apparently the `axis` parameter is only available to ufuncs with a "single shared core dimension".
It's not 100% clear to me what exactly this means and how to extract this info from the ufuncs signature.
If anyone could provide some insight here then this could help with conditionally removing `axis`.

Reply:
@seberg Thoughts?

Reply:
In theory... you could parse the signature to see if there is only a single distinct letter included.  The ufunc does store the number (that has to be 1) explicitly on the C-side, but doens't expose it.

But I guess you don't have access to that runtime information, so probably just have to allow this.  An example that fails is `matmul`.

Reply:
Ah so be it, false negatives are generally less disruptive than false positives anyway.